cfme/infrastructure/networking.py

```diff
@@ -1,73 +1,118 @@
 import attr
 from navmazing import NavigateToAttribute
+from widgetastic.widget import Text
 from widgetastic.widget import View
 from widgetastic_patternfly import Dropdown
 
 from cfme.base.ui import BaseLoggedInPage
+from cfme.common import CustomButtonEventsMixin
+from cfme.common import Taggable
 from cfme.modeling.base import BaseCollection
 from cfme.modeling.base import BaseEntity
 from cfme.utils.appliance.implementations.ui import CFMENavigateStep
+from cfme.utils.appliance.implementations.ui import navigate_to
 from cfme.utils.appliance.implementations.ui import navigator
+from widgetastic_manageiq import Accordion
+from widgetastic_manageiq import BaseEntitiesView
 from widgetastic_manageiq import ItemsToolBarViewSelector
-from widgetastic_manageiq import PaginationPane
-from widgetastic_manageiq import Text
+from widgetastic_manageiq import ManageIQTree
+from widgetastic_manageiq import SummaryTable
 
 
-class InfraNetworking(BaseEntity):
-    _param_name = 'InfraNetworking'
+@attr.s
+class InfraSwitches(BaseEntity, Taggable, CustomButtonEventsMixin):
+    """ Model of an infrastructure cluster in cfme
+
+    Args:
+        name: Name of the switch.
+    """
+
+    name = attr.ib()
 
 
 @attr.s
-class InfraNetworkingCollection(BaseCollection):
+class InfraSwitchesCollection(BaseCollection):
     """Collection object for the :py:class:`cmfe.infrastructure.networking.InfraNetworking`."""
-    ENTITY = InfraNetworking
+
+    ENTITY = InfraSwitches
+
+    def all(self):
+        """List of all switch objects"""
+        view = navigate_to(self, "All")
+        return [self.instantiate(ent) for ent in view.entities.entity_names]
 
 
 class InfraNetworkingView(BaseLoggedInPage):
     """Base view for header and nav checking, navigatable views should inherit this"""
 
     @property
     def in_infra_networking(self):
```

Comment:
Question: Just curious, why is this not an `is_displayed` property?

Reply:
@john-dupuy: 
1. Generally, we put entry point view for model pages best word `BaseView` for respective model page.
All other view we try to create by inheritance. Here this view not holding `widgets`. It is only for check that we are on that `BaseView` or not by using vertical nav tree and user login. so better to call `in_model_page`.
2. In case we put this as `is_displayed`; then for every view if we want robust check then we need check specifically we are satisfying `vertical nav tree` and `logged as current user` in `is_displayed`. 

Reference:
https://github.com/ManageIQ/integration_tests/pull/5705#discussion_r148034562